features/svc-catalog_asb/svc-catalog.feature

```diff
@@ -207,6 +207,152 @@ Feature: Service-catalog related scenarios
+    And I use the "<%= cb.ups_broker_project %>" project
+    When I process and create:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/svc-catalog/ups-broker-template.yaml |
+      | p | UPS_BROKER_PROJECT=<%= cb.ups_broker_project %>                                                         |
+    Then the step should succeed
+    And I wait up to 300 seconds for the steps to pass:
+    """
+    When I run the :describe client command with:
+      | resource | clusterservicebroker/ups-broker |
+    Then the output should contain "Successfully fetched catalog entries from broker"
+    When I run the :get client command with:
+      | resource | clusterserviceclass                                                       |
+      | o        | custom-columns=CLASSNAME:.metadata.name,EXTERNAL\ NAME:.spec.externalName |
+    Then the output should contain "user-provided"
+    """
+
+    #Provision a serviceinstance
+    Given I switch to the first user
+    And I use the "<%= cb.user_project %>" project
+    When I process and create:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/svc-catalog/ups-instance-template.yaml |
+      | p | USER_PROJECT=<%= cb.user_project %>                                                                       |
+    Then the step should succeed
+    And I wait up to 30 seconds for the steps to pass:
+    """
+    When I run the :describe client command with:
+      | resource | serviceinstance/ups-instance |
+    Then the output should match "Message.*The instance was provisioned successfully"
+    """
+
+    # Create servicebinding
+    When I process and create:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/svc-catalog/ups-binding-template.yaml |
+      | p | USER_PROJECT=<%= cb.user_project %>                                                                      |
+    Then the step should succeed
+    Given I check that the "my-secret" secret exists
+    And I wait up to 10 seconds for the steps to pass:
+    """
+    When I run the :describe client command with:
+      | resource | servicebinding |
+    Then the output should match "Message.*Injected bind result"
+    """
+
+    #Update servicebinding
+    When I run the :patch client command with:
+      | resource | servicebinding/ups-binding                     |
+      | p        | {"metadata":{"labels":{"app":"test-binding"}}} |
+    Then the step should succeed
+    When I run the :get client command with:
+      | resource   | servicebinding/ups-binding |
+      | show_label | true                       |
+    Then the output should contain "app=test-binding"
+
+    # Delete servicebinding
+    When I run the :delete client command with:
+      | object_type       | servicebinding |
+      | object_name_or_id | ups-binding    |
+    Then the step should succeed
+    Given I ensure "ups-binding" servicebinding is deleted
```

Comment:
the `I ensure` step can replace the line above (349-352)

Reply:
Updated